catwalk/models/huggingface.py

```diff
@@ -18,19 +19,80 @@ class HFAutoModel(Model):
 
     def __init__(self, pretrained_model_name_or_path: str):
         self.pretrained_model_name_or_path = pretrained_model_name_or_path
-
+        
     def predict(  # type: ignore
         self,
         task: Task,
         instances: Sequence[Dict[str, Any]],
         *,
         batch_size: int = 32
     ) -> Iterator[Dict[str, Any]]:
-        if not task.has_instance_conversion(InstanceFormat.HF_MC):
-            raise UnsupportedTaskError(self, task)
-        instances = MappedSequence(
-            lambda instance: task.convert_instance(instance, InstanceFormat.HF_MC),
-            instances)
+        if task.has_instance_conversion(InstanceFormat.HF_MC):
+            return self._predict_mc(task, instances, batch_size=batch_size)
+        elif task.has_instance_conversion(InstanceFormat.HF_QA):
+            return self._predict_qa(task, instances, batch_size=batch_size)
+        
+        raise UnsupportedTaskError(self, task)
+
+    
+    def _convert_instances(self, instances: Sequence[Dict[str, Any]], instance_format, task) -> MappedSequence:
+        return MappedSequence(lambda instance: task.convert_instance(instance, instance_format), instances)
+    
+    def _predict_qa(
+        self,
+        task: Task,
+        instances: Sequence[Dict[str, Any]],
+        batch_size: int = 32
+    ) -> Iterator[Dict[str, Any]]:
+        instances = self._convert_instances(instances, InstanceFormat.HF_QA, task)
+        
+        device = resolve_device()
+        model = cached_transformers.get(AutoModelForQuestionAnswering, self.pretrained_model_name_or_path, False).eval().to(device)
```

Comment:
This is good, but why didn't you use [QuestionAnsweringPipeline](https://huggingface.co/docs/transformers/v4.20.1/en/main_classes/pipelines#transformers.QuestionAnsweringPipeline)?

Reply:
A limitation of using pipelines is that it doesn't support distributed evaluation if you want to add this later on. If this is not something you want to add then we can use pipelines and remove the post-processing overhead.

Reply:
Will we be able to load up a big model, like the BLOOM model, which doesn't fit on a single GPU, and run evaluations? Are pipelines a torch-only proposition, or can we run Jax models with them?

Reply:
It seems that pipelines currently only support PyTorch and Tensorflow. I played around with [this feature] (https://huggingface.co/docs/transformers/v4.20.1/en/main_classes/model#large-model-loading) and it seems like it can be used inside pipelines.